vllm/README.md

```diff
@@ -0,0 +1,206 @@
+# vLLM Truss
+
+## What is this Truss example doing
+
+This is a generic [Truss](https://truss.baseten.co/) that can deploy an asynchronous vLLM engine([AsyncLLMEngine](https://docs.vllm.ai/en/latest/dev/engine/async_llm_engine.html#asyncllmengine)) of any customized configuration with [all compatible models](https://docs.vllm.ai/en/latest/models/supported_models.html). We create this example to give you the most codeless experience, so you can configure all vLLM engine parameters in `config.yaml`, without making code changes in `model.py` for most of the use cases.
+
+## Configure your Truss by modifying the config.yaml
+
+### Basic options using 1 GPU
+
+Here is the minimum config file you will need to deploy a model using vLLM on 1 GPU.
+The only parameters you need to touch are:
+- `model_name`
+- `repo_id`
+- `accelerator`
+
+```
+model_name: "Llama 3.1 8B Instruct VLLM"
+python_version: py311
+model_metadata:
+  example_model_input: {"prompt": "what is the meaning of life"}
+  repo_id: meta-llama/Meta-Llama-3.1-8B-Instruct
+  openai_compatible: true
+  vllm_config: null
+requirements:
+  - vllm==0.5.4
+resources:
+  accelerator: A100
+  use_gpu: true
+runtime:
+  predict_concurrency: 128
+secrets:
+  hf_access_token: null
+```
+
+### Basic options using multiple GPUs
+
+If your model needs more than 1 GPU to run using tensor parallel, you will need to change `accelerator`, and to set `tensor_parallel_size` and `distributed_executor_backend` accordingly.
+
+```
+model_name: "Llama 3.1 8B Instruct VLLM"
+python_version: py311
+model_metadata:
+  example_model_input: {"prompt": "what is the meaning of life"}
+  repo_id: meta-llama/Meta-Llama-3.1-8B-Instruct
+  openai_compatible: false
+  vllm_config:
+    tensor_parallel_size: 4
+    max_model_len: 4096
+    distributed_executor_backend: mp
```

Comment:
what do you think about optionally defaulting to some backend in model.py if this isn't configured explicitly?

Reply:
by default vllm uses "ray" if installed otherwise "mp"